core/textile/model/mirror_file.go

```diff
@@ -0,0 +1,142 @@
+package model
+
+import (
+	"context"
+	"errors"
+
+	"github.com/FleekHQ/space-daemon/core/space/domain"
+	"github.com/FleekHQ/space-daemon/log"
+	"github.com/textileio/go-threads/api/client"
+	core "github.com/textileio/go-threads/core/db"
+	"github.com/textileio/go-threads/core/thread"
+	"github.com/textileio/go-threads/db"
+	"github.com/textileio/go-threads/util"
+)
+
+type MirrorFileSchema struct {
+	ID         core.InstanceID `json:"_id"`
+	Path       string          `json:"path"`
+	BucketSlug string          `json:"bucket_slug"`
+	Backup     bool            `json:"backup"`
+	Shared     bool            `json:"shared"`
+
+	DbID string
+}
+
+type MirrorBucketSchema struct {
+	RemoteDbID string `json:"remoteDbId"`
+	HubAddr    string `json:"HubAddr"`
+}
+
+const mirrorFileModelName = "MirrorFile"
+
+var errMirrorFileNotFound = errors.New("Mirror file not found")
+
+func (m *model) CreateMirrorBucket(ctx context.Context, bucketSlug string, mirrorBucket *MirrorBucketSchema) (*BucketSchema, error) {
+	metaCtx, metaDbID, err := m.initBucketModel(ctx)
+	if err != nil && metaDbID == nil {
+		return nil, err
+	}
+
+	bucket, err := m.FindBucket(ctx, bucketSlug)
+	if err != nil {
+		return nil, err
+	}
+
+	bucket.RemoteDbID = mirrorBucket.RemoteDbID
+	bucket.HubAddr = mirrorBucket.HubAddr
+
+	instances := client.Instances{bucket}
+
+	err = m.threads.Save(metaCtx, *metaDbID, bucketModelName, instances)
+	if err != nil {
+		return nil, err
+	}
+
+	return bucket, nil
+}
+
+// Finds the metadata of a file that has been shared to the user
+func (m *model) FindMirrorFileByPathAndBucketSlug(ctx context.Context, path, bucketSlug string) (*MirrorFileSchema, error) {
```

Comment:
`bucketSlug` is not used here. I think it doesn't cause any issues since the bucket slug is always `personal` but maybe we can add that to the `Find` query just in case.